### PR TITLE
D1 batch queries

### PR DESCRIPTION
**Pull request by ejyager00**
@G4brym This is a pretty big refactor and includes breaking changes, but I saw a lot of people were interested in the batch query possibilities of D1, and I thought this might be a way to implement it. Would you be interested in these changes, perhaps as a major release? Alternatively, if you have another idea for how to implement this functionality, I'm open to suggestions.

### CODE REVIEW

**Comment by G4brym** [comment]
Hey @ejyager00 thanks for doing this work 😮 
I will need some days to review this

File: src/databases/d1.ts [hunk 1]
```diff
@@ -11,36 +11,92 @@ export class D1QB extends QueryBuilder<D1Result, D1ResultOne> {
...
         success: resp.success,
-        results: params.fetchType === FetchTypes.ONE && resp.results.length > 0 ? resp.results[0] : resp.results,
+        results: query.fetchType === FetchTypes.ONE && resp.results.length > 0 ? resp.results[0] : resp.results,
       }
     }
 
     return stmt.run()
   }
+
+  async batchExecute(queryArray: Query[]): Promise<(D1ResultOne | D1Result)[]> {
```
Comment (G4brym) [inline src/databases/d1.ts:47]: can you add a simple test for this function

File: src/databases/d1.ts [hunk 2]
```diff
@@ -11,36 +11,92 @@ export class D1QB extends QueryBuilder<D1Result, D1ResultOne> {
...
 
-    if (params.fetchType === FetchTypes.ONE || params.fetchType === FetchTypes.ALL) {
+    if (query.fetchType === FetchTypes.ONE || query.fetchType === FetchTypes.ALL) {
       const resp = await stmt.all()
 
       return {
         changes: resp.meta?.changes,
         duration: resp.meta?.duration,
-        last_row_id: resp.meta?.last_row_id,
+        lastRowId: resp.meta?.last_row_id,
```
[the remark this answers is not in the document]
Comment (ejyager00) [reply src/databases/d1.ts]: I updated it to the snake case version, was that what you had intended? If not, I can switch to camel case.

**Review by G4brym — APPROVED** [review]

**Comment by G4brym** [comment]
Hey @ejyager00 thanks for making these changes, i will update the docs and open a major release next week

**Pull request merged**